### ISSUE TITLE
error! macro collides with log's error! macro

### ISSUE THREAD

**Issue by partim** [Geal/nom#169]
Both nom and log have an `error!` macro.

Since log is a rather fundamental crate, perhaps it would be good to rename nom’s version? In the spirit of `try_parse!`, perhaps `parse_error!`?

**Comment by Geal** [comment]
This is something I'm considering for nom 2.0, since it will break the current API.

In the meantime, you can selectively import macros like this:

```rust
#[macro_use(many0, named)]
extern crate nom;
```

It would be great if macros could have a crate prefix.

**Comment by aldanor** [comment]
@Geal ... and you have to also bring in all the internal macros, otherwise it wouldn't work -- e.g. if you want to use `map!`, you have to do something like:

```rust
#[macro_use(map, map_impl, call)]
extern crate nom;
```

**Comment by aldanor** [comment]
A real usage example :cry: :panda_face: 

```rust
#[macro_use(map, map_impl, map_res, map_res_impl, try_parse, call, named, chain, tag_bytes)]
#[macro_use(tuple, tuple_parser, terminated, separated_pair, char, tag, chaining_parser)]
extern crate nom;
```

**Comment by Geal** [comment]
oh, that's really awful :/

Replacing the current `error!` macro would require a major version number increase, and would break compatibility with existing code. But I think I could make a compilation feature to use `error!` in default case, and another name if the feature is activated?

**Comment by partim** [comment]
Perhaps you could expose it additionally under a new name for now? That should only require a minor bump. By importing nom first and log later, all would be good.

**Comment by Geal** [comment]
that could work too

**Comment by aldanor** [comment]
Yea, either way (or both) would be good! Perhaps that's something that could go into README as well so people are not confused.

**Comment by Geal** [comment]
this is now available in nom 2.0
